### DISCUSSION TITLE
Changelog Announcements

### DISCUSSION THREAD

**Discussion by danisyellis** [indeedeng/starfish#100] [Show and tell]
This is a log of important changes made to the Starfish repo.
We may not list all changes here, but I'll always make sure to put in info about breaking changes and security updates when those changes become live.
If you're using Starfish, I'd recommend Subscribing to Notifications (using the button on the right-hand side of this page) so that you'll be notified when there are new, exciting features or fixes for security vulnerabilities.

**Comment by danisyellis** [comment]
**SECURITY UPDATE (FOR DEVELOPERS)**
The node-notifier package has been updated to fix a security vulnerability. 
This package is used by Jest, so it's not an issue for anyone running Starfish normally.
It is an issue for developers who run the tests. (`npm test`)
If you are doing dev work on Starfish and don't already have the latest code, you should pull in the latest code and run `npm ci` in your terminal to update all packages, before running the Starfish tests.

**Comment by IngridGdesigns** [comment]
**Dependabot Vulnerability Bump y18n from 4.0.0 to 4.0.1 Update**

This vulnerability would only affect the use of locale, which Starfish is not using, but we still updated. To read further more about the Bug Fixes[ visit the y18n changelog](https://github.com/yargs/y18n/blob/master/CHANGELOG.md#401-2020-10-25)

**Comment by danisyellis** [comment]
### Renamed Master to Main, and 2 security fixes

We've renamed the default branch from Master to Main. GitHub has made this really easy to do. You can read how [Here](https://github.com/github/renaming)
You'll need to update your local repo before pushing or pulling. If you go to Starfish's [homepage](https://github.com/indeedeng/starfish) you should see a message from GitHub explaining how. Just in case, I'll paste the instructions here:
```
The default branch has been renamed!
master is now named main
If you have a local clone, you can update it by running:
git branch -m master main
git fetch origin
git branch -u origin/main main
git remote set-head origin -a
```

Also, once you update that, I'd recommend pulling in the most recent changes. We just did two dependency updates because of security vulnerabilities.

**Comment by danisyellis** [comment]
Hello Starfisherpeople.

When you get a chance, pull in the latest code. We've patched a few security vulnerabilities that popped up recently. 

We also did a good-sized code refactor so if you want to contribute to the project, the code is a lot easier to read now and somewhat easier to code on. Just in time for Hacktoberfest! 

If you're a Starfish user, you'll only notice one change, but it's important. Once you pull in the new code (and run `npm ci`), to run Starfish, instead of typing into the terminal `node index.js < {path/to/CSVfile}.csv {date1} {date2}` pieces of that command have been rearranged a bit. Now you'll use `node index.js {date1} {date2} {path/to/CSVfile}.csv`. If anything about this is confusing, check out the README for more info. And if that doesn't help, create a new Discussion with your questions.

Since changing the command you use to run the project is a breaking change, we're going to call this version 3.0!
